index/finder.go

```diff
@@ -45,72 +55,69 @@ func NewFinder(ctx context.Context, zctx *zed.Context, engine storage.Engine, ur
 	}, nil
 }
 
-type operator int
-
-const (
-	eql operator = iota
-	gte
-	lte
-)
-
 // lookup searches for a match of the given key compared to the
 // key values in the records read from the reader.  If the op argument is eql
 // then only exact matches are returned.  Otherwise, the record with the
 // largest key smaller (or larger) than the key argument is returned.
-func lookup(reader zio.Reader, compare expr.KeyCompareFn, o order.Which, op operator) (*zed.Value, error) {
+func lookup(reader zio.Reader, compare expr.KeyCompareFn, o order.Which, op Operator) (*zed.Value, error) {
 	if o == order.Asc {
 		return lookupAsc(reader, compare, op)
 	}
 	return lookupDesc(reader, compare, op)
 }
 
-func lookupAsc(reader zio.Reader, fn expr.KeyCompareFn, op operator) (*zed.Value, error) {
+func lookupAsc(reader zio.Reader, fn expr.KeyCompareFn, op Operator) (*zed.Value, error) {
 	var prev *zed.Value
 	for {
 		rec, err := reader.Read()
 		if rec == nil || err != nil {
-			if op == eql || op == gte {
+			if op == EQL || op == GTE || op == GT {
 				prev = nil
 			}
 			return prev, err
 		}
-		if cmp := fn(rec); cmp >= 0 {
-			if cmp == 0 {
+		cmp := fn(rec)
+		if cmp >= 0 {
+			if cmp == 0 && op.hasEqual() {
 				return rec, nil
 			}
-			if op == eql {
+			if op == LTE || op == LT {
+				return prev, nil
+			}
+			if op == EQL {
 				rec = nil
 			}
-			if op == lte {
-				return prev, nil
+			if !(op == GT && cmp == 0) {
+				return rec, nil
 			}
-			return rec, nil
 		}
 		prev = rec
 	}
 }
 
-func lookupDesc(reader zio.Reader, fn expr.KeyCompareFn, op operator) (*zed.Value, error) {
+func lookupDesc(reader zio.Reader, fn expr.KeyCompareFn, op Operator) (*zed.Value, error) {
 	var prev *zed.Value
 	for {
 		rec, err := reader.Read()
 		if rec == nil || err != nil {
-			if op == eql || op == lte {
+			if op == EQL || op == LTE || op == LT {
 				prev = nil
 			}
 			return prev, err
 		}
 		if cmp := fn(rec); cmp <= 0 {
-			if cmp == 0 {
+			if cmp == 0 && op.hasEqual() {
 				return rec, nil
 			}
-			if op == eql {
-				rec = nil
-			}
-			if op == gte {
+			if op == GTE || op == GT {
 				return prev, nil
 			}
-			return rec, nil
+			if op == EQL {
+				return nil, nil
```

Comment:
Should this case remain consistent with the corresponding case in lookupAsc? (They were previously identical but now diverge.)

Reply:
They are the same thing but I'll change the one above to this.